src/identifiers.rs

```diff
@@ -227,8 +222,11 @@ impl RoomId {
 
     /// Returns the server name of the room ID.
     #[napi(getter)]
-    pub fn server_name(&self) -> ServerName {
-        ServerName { inner: self.inner.server_name().unwrap().to_owned() }
+    pub fn server_name(&self) -> Option<ServerName> {
```

Comment:
Can we bin this function?  It's absolute nonsense in *all* room versions to pull out the server name from rooms, and we only ever did this as a hack on what is an opaque string.

What is using it?

Reply:
Nothing in https://github.com/element-hq/matrix-bot-sdk, at least.